src/main/java/no/unit/nva/bare/AddNewAuthorityIdentifierHandler.java

```diff
@@ -14,63 +17,59 @@
+            "Communication failure trying to update authority";
     public static final String SCN_KEY = "scn";
     public static final String QUALIFIER_KEY = "qualifier";
-    public static final String IDENTIFIER_KEY = "identifier";
-    public static final int ERROR_CALLING_REMOTE_SERVER = Response.Status.BAD_GATEWAY.getStatusCode();
     public static final String REMOTE_SERVER_ERRORMESSAGE = "remote server errormessage: ";
 
     public static final List<String> VALID_QUALIFIERS = asList(ValidIdentifierKey.FEIDEID.asString(),
             ValidIdentifierKey.ORCID.asString(), ValidIdentifierKey.ORGUNITID.asString());
 
-    protected final transient BareConnection bareConnection;
-    private final transient Logger log = Logger.instance();
-
+    private transient BareConnection bareConnection;
 
+    /**
+     * Default constructor for AddNewAuthorityIdentifierHandler.
+     */
+    @JacocoGenerated
     public AddNewAuthorityIdentifierHandler() {
-        this.bareConnection = new BareConnection();
-    }
-
-    public AddNewAuthorityIdentifierHandler(BareConnection bareConnection) {
-        this.bareConnection = bareConnection;
+        this(new Environment(), new BareConnection());
     }
 
     /**
-     * Main lambda function to add a qualified identifier on an existing Bare authority.
+     * Constructor for AddNewAuthorityIdentifierHandler.
      *
-     * @param input payload with path-parameters
-     * @return a GatewayResponse
+     * @param environment    environment
+     * @param bareConnection bareConnection
      */
+    public AddNewAuthorityIdentifierHandler(Environment environment, BareConnection bareConnection) {
+        super(AddNewAuthorityIdentifierRequest.class, environment,
+                LoggerFactory.getLogger(AddNewAuthorityIdentifierHandler.class));
+        this.bareConnection = bareConnection;
+    }
+
     @Override
-    @SuppressWarnings("unchecked")
-    public GatewayResponse handleRequest(final Map<String, Object> input, Context context) {
-        GatewayResponse gatewayResponse = new GatewayResponse();
-        try {
-            this.checkParameters(input);
-        } catch (RuntimeException e) {
-            log.error(e);
-            gatewayResponse.setErrorBody(e.getMessage());
-            gatewayResponse.setStatusCode(Response.Status.BAD_REQUEST.getStatusCode());
-            return gatewayResponse;
-        }
-        Map<String, String> pathParameters = (Map<String, String>) input.get(PATH_PARAMETERS_KEY);
-        String scn = pathParameters.get(SCN_KEY);
-        String inputQualifier = pathParameters.get(QUALIFIER_KEY);
+    protected Authority processInput(AddNewAuthorityIdentifierRequest input, RequestInfo requestInfo,
+                                     Context context) throws ApiGatewayException {
+
+        validateInput(input, requestInfo.getPathParameters());
+
+        String scn = requestInfo.getPathParameters().get(SCN_KEY);
```

Comment:
RequestUtil.getPathParameter(requestInfo, SCN_KEY)

Håndterer feil for deg.

Reply:
Nice, I didnt notice yesterdays new version, will bump commons to 0.3.5 and make use of it!

Reply:
Using v0.3.5 and getPAthParameter